cmd/helm/load_plugins.go

```diff
@@ -170,7 +170,9 @@ func manuallyProcessArgs(args []string) ([]string, []string) {
 				return v
 			}
 		}
-		return ""
+		// Return something different than what we received by adding a random prefix.
+		// We cannot simply return an empty string as v could be an empty string itself.
+		return fmt.Sprintf("x%s", v)
```

Comment:
I don't think this is the right place to make the change. I think there are some unintended consequences though I've not worked out a better way to solve this problem.

`isKnown` is used to see if something is part of the above `kargs`. In the switch statement below this triggers some logic. With this change to `isKnown` the default section of the `switch` statement below will never be executed. All arguments will be labeled as known and unknown will always be empty.

Reply:
Thanks @mattfarina for reviewing this.

It's been a long time but looking at the switch statement, I read the second `case` as equivalent to `if a == isKnown(a)`.  Or am I wrong?

If that is the case, then the requirement is that `isKnown()` return anything different than `a` if `a` is not part of `kvargs`, so returning `a` with a prefix of `x` satisfies this requirement, no?

But I might be misreading something.

Reply:
Pre-empting this comment with I am very new to this code base and this can be treated as me thinking out loud. 

I did some testing by debugging the `TestManuallyProcessArgs` test and I believe this assumption `if a == isKnown(a)` is correct, strings like `""` and `" "` or even `"x"` end up in the unknown arg list. 

That said I do think the switch statement is limiting in this scenario since it causes you to handle edge cases by modifying `IsKnown` rather than allowing you to add an explicit check for this scenario. Converting the switch statement into if statements  would allow you to avoid the need to add a prefix to the arg just so it ends up in the unknown arg list.